src/librepgp/stream-armour.c

```diff
@@ -493,24 +493,27 @@ find_armour_header(const char *buf, size_t len, size_t *hdrlen)
     return NULL;
 }
 
-static pgp_armoured_msg_t
-armour_message_type(const char *hdr, size_t len)
+pgp_armoured_msg_t
+armour_str_to_data_type(const char *str, size_t len)
 {
-    if (!strncmp(hdr, "BEGIN PGP MESSAGE", len)) {
+    if (!str) {
+        return PGP_ARMOURED_UNKNOWN;
+    }
+
+    if (!strncmp(str, "BEGIN PGP MESSAGE", len) || !strncmp("msg", str, len)) {
         return PGP_ARMOURED_MESSAGE;
-    } else if (!strncmp(hdr, "BEGIN PGP PUBLIC KEY BLOCK", len) ||
-               !strncmp(hdr, "BEGIN PGP PUBLIC KEY", len)) {
+    } else if (!strncmp(str, "BEGIN PGP PUBLIC KEY BLOCK", len) ||
```

Comment:
We can shortcut all of these `else if` Into just `if` since they are all returned.